server/game/cards/07-GR/JackpotGreene.js

```diff
@@ -0,0 +1,32 @@
+const Card = require('../../Card.js');
+
+class JackpotGreene extends Card {
+    // After a creature reaps, discard the top card of its
+    // controller's deck. If the discarded card is a Star Alliance
+    // card, you gain 1.
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            targetController: 'any',
+            match: (card) => card.type === 'creature',
+            effect: ability.effects.gainAbility('reap', {
+                condition: (context) => context.source.controller.deck.length > 0,
+                gameAction: ability.actions.discard((context) => ({
+                    target: context.source.controller.deck[0]
+                })),
+                then: {
+                    condition: (context) =>
+                        context.preThenEvents &&
+                        context.preThenEvents.length > 0 &&
+                        context.preThenEvents[0].card.hasHouse('staralliance'),
+                    gameAction: ability.actions.gainAmber({
+                        target: this.controller
```

Comment:
I know we're supposed to avoid using `this` but I can't find another way to get the controller of Jackpot Greene inside this reap ability.  Let me know if I'm missing something.

Reply:
`context.source.controller` ?

Reply:
Unfortunately not.  After the opponent creature reaps, the `context.source` is that creature, not Jackpot Greene anymore.  Not sure if that's a bug in the engine or not but that's how it works in practice. (In fact, in the `discard` action above, I rely on that to discard from the correct deck.)

Reply:
You could try building `then` dynamically :

`then: (preThenContext) => {`

and use `preThenContext.source.controller`

Check the code for examples.

Reply:
That didn't work; the preTheContext source is also the creature.

But I decided to just turn this whole thing into a reap reaction, rather than a reap ability (which is more correct to the card text anyway), and now it works just fine without using `this`.  Please take another look!